src/public/WRLDRouteStep.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#import <Foundation/Foundation.h>
+#import <CoreLocation/CoreLocation.h>
+
+#import "WRLDRouteDirections.h"
+#import "WRLDRouteTransportationMode.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/*!
+ A single step of a WRLDRoute.
+ */
+@interface WRLDRouteStep : NSObject
+
+/*!
+ An array of the individual CLLocationCoordinate2D points that make up this step.
+ This can be a single point if no distance was covered, for example
+ a WRLDRouteStep may indicate departure or arrival with a single point.
+ */
+ @property (nonatomic) CLLocationCoordinate2D* path;
+
+/*!
+ The count of CLLocationCoordinate2D points that make up this path.
+ */
+@property (nonatomic) int pathCount;
+
+/*!
+ The directions associated with this step.
+ */
+@property (nonatomic) WRLDRouteDirections* directions;
+
+/*!
+ Specifies the mode of transport for this step:
+ - `WRLDWalking`: Indicates that the route is a walking WRLDRoute.
+ */
+@property (nonatomic) WRLDRouteTransportationMode mode;
+
+/*!
+ Whether this step is indoors or not.
+ */
+@property (nonatomic) BOOL isIndoors;
+
+/*!
+ If indoors, the ID of the indoor map this step is inside.
+ */
+@property (nonatomic, copy) NSString* indoorId;
+
+/*!
+ If indoors, the ID of the floor this step is on.
+ */
+@property (nonatomic) int indoorFloorId;
+
+/*!
+ The estimated time this step will take to travel in seconds.
+ */
+@property (nonatomic) double duration;
```

Comment:
Minor thing: In other parts of the API (e.g animate camera view) we specify duration as an NSTimeInterval - might be some mileage in doing the same here, but happy enough since the units are specified in the comments.

Reply:
Updated so it uses NSTimeInterval instead of double.
https://github.com/wrld3d/ios-api/pull/23/commits/adf5eac61f922029ffafb8db2eaefd4ed3475820